sys/kern/exec.c

```diff
@@ -284,7 +284,7 @@ static void enter_new_vmspace(proc_t *p, exec_vmspace_t *saved,
   vm_object_t *stack_obj = vm_object_alloc(VM_ANONYMOUS);
   vm_segment_t *stack_seg =
     vm_segment_alloc(stack_obj, USER_STACK_TOP - USER_STACK_SIZE,
-                     USER_STACK_TOP, VM_PROT_READ | VM_PROT_WRITE);
+                     USER_STACK_TOP, VM_PROT_READ | VM_PROT_WRITE, VM_FILE);
```

Comment:
`VM_FILE` here? We don't have file mappings! Why not `VM_ANON`? Similarly in many other places...